src/evaluation/factuality.py

```diff
@@ -97,82 +162,61 @@ def evaluate_factuality(
         "rouge2": [],
         "rougeL": [],
         "sum_with_extrinsic": 0,
+        "count_entity_label": defaultdict(lambda: 0)
     }
-    for value in ANNOTATION_LABELS.values():
-        counters[value] = 0
 
     print_counter = 0
     for sum_id, summary in tqdm(sorted(sums_by_id.items(), key=lambda x: x[0])):
-        source = xsum_test[sum_id]["document"]
-        reference = gold_sums[sum_id]["summary"]
+        summary_eval = evaluate_summary(
+            summary,
+            xsum_test[sum_id]["document"],
+            gold_sums[sum_id]["summary"],
+            labeled_ents[sum_id],
+            is_fbs,
+            is_oracle,
+        )
+        count_non_factual_extrinsic = summary_eval["count_entity_label"][
+            ANNOTATION_LABELS["Non-factual"]
+        ]
+        count_non_factual_intrinsic = summary_eval["count_entity_label"][
+            ANNOTATION_LABELS["Intrinsic"]
+        ]
 
+        # Init the possible eval states of a summary
         non_factual_extrinsic = False
         non_factual_intrinsic = False
-        is_skipped = False
-        has_unknown = False
-        has_failed = False
+        non_factual = False
+        has_unknown = (
+            summary_eval["count_entity_label"][ANNOTATION_LABELS["Unknown"]] > 0
+        )
+        count_extrinsic = 0
 
-        if summary == SUMMARY_FAILED_GENERATION:
+        # Skip logic
+        is_skipped = False
+        if summary_eval["failed"]:
+            is_skipped = True
+        elif is_fbs and is_oracle and count_non_factual_extrinsic:
```

Comment:
Is it possible for FBS in oracle-model to have a more similar API to non-oracle FBS: aka, could it return `SUMMARY_FAILED_GENERATION` as well?

Reply:
Hmm, they both have the same API - `SUMMARY_FAILED_GENERATION` means that the summary generation failed, i.e. no valid beams were left, for both FBS and oracle.

The case you're looking at here is when a summary was generated, but an entity was non-factual. This indicates a skip, but the summary is still returned from the model (it's not the same as `SUMMARY_FAILED_GENERATION`).

Reply:
makes sense!